README.md

```diff
@@ -14,148 +14,498 @@ referred to as AuthZ) and is what we'll discuss in this section.
+    that is how this table and the `user_authorities` table is joined.
+  - The `inverseJoinColumns` assigns the column of the other table related to
+    the associated entity, or in this case, the `authorities` table. The
+    foreign key that links these tables together is the `authority_id`, so
+    that will be specified as the `inverseJoinColumns`.
+
+Now that we have our entities define, what do we do now?
+
+We need to define the authority that the users have! Remember in the last lesson
+we said we would come back to the `getAuthorities()` method in our `UserWrapper`
+class? It's time to do so!
+
+Now that we have our authority entity, we can use this in the `getAuthorities()`
+method... but there is one catch. We need to return a collection of
+`GrantedAuthoriy` instances. So we'll use the same design pattern we did in the
+last lesson which is to create a wrapper class that encompasses an `Authority`
+instance and implements the `GrantedAuthority` interface. Within the `entity`
+package, create a `AuthorityWrapper` class that implements the
+`GrantedAuthority` interface:
+
+```java
+package com.example.springsecuritydemo.entity;
 
-        return userDetailService;
+import org.springframework.security.core.GrantedAuthority;
+
+public class AuthorityWrapper implements GrantedAuthority {
+    
+    private final Authority authority;
+    
+    public AuthorityWrapper(Authority authority) {
+        this.authority = authority;
+    }
+    
+    @Override
+    public String getAuthority() {
+        return authority.getName();
     }
+}
 ```
 
-As you can see, we have given our `user1` the "read" authority and our
-`adminUser1` the "admin" authority. Note that these 2 authorities are arbitrary
-and only matter in that they will need to match with the AuthZ rules we're about
-to set in our `SecurityConfiguration` `configure` method:
+In this class, we'll need to override the `getAuthority()` method. We can do
+this by returning the authority's name.
+
+Now let's go back to our `UserWrapper` class and look at the `getAuthorities()`
+method again:
 
 ```java
     @Override
-    protected void configure(HttpSecurity http) throws Exception {
-        http.authorizeRequests()
-                .antMatchers("/status")
-                .hasAuthority("admin");
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return List.of(() -> "read");
+    }
+```
 
-        http.authorizeRequests()
-                .anyRequest()
-                .authenticated()
-                .and()
-                .formLogin();
+We'll modify this method to reflect our authorities stored in our database that
+are associated with our user. Consider the following edits to this method:
 
+```java
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return user.getAuthorities().stream().map(AuthorityWrapper::new).collect(Collectors.toList());
```

Comment:
At this point in the curriculum, have students learned streams and method references?

Reply:
Yes! That is the assumption that in Java Mod 4 they will have learned about all of this - @linda-seiter, correct me if I'm wrong.

Reply:
Correct, module 4 covers streams and method references